### PR TITLE
Rebased #831

### PR DESCRIPTION
**Pull request by lu-zero**

### CODE REVIEW

**Comment by rzumer** [comment]
Just FYI, by default you should be able to push such a change directly to the #831 branch.

**Comment by lu-zero** [comment]
```
test test_encode_decode::high_bd ... FAILED
```

and just this...

File: src/partition.rs [hunk 1]
```diff
@@ -899,111 +899,73 @@ pub enum MvJointType {
...
         let left_slice = dst.go_left(1);
-        for i in 0..B::H {
-          left[2*MAX_TX_SIZE - B::H + i] = left_slice.p(0, B::H - 1 - i);
+        for i in 0..tx_size.height() {
+          left[2*MAX_TX_SIZE - tx_size.height() + i] = left_slice.p(0, tx_size.height() - 1 - i);
         }
       } else {
         let val = if y != 0 { dst.go_up(1).p(0, 0) } else { base + 1 };
-        for v in left[2*MAX_TX_SIZE - B::H..].iter_mut() {
+        for v in left[2*MAX_TX_SIZE - tx_size.width()..].iter_mut() {
```
Comment (ycho) [inline src/partition.rs]: This should be: tx_size.height().

Comment (lu-zero) [reply src/partition.rs]: updated accordingly :)

**Pull request merged**